LuaUI/cawidgets.lua

```diff
@@ -1997,11 +2017,24 @@ end
 
 
 function widgetHandler:PlayerChanged(playerID) --when player Change from Spectator to Player or Player to Spectator.
-  MessageProcessor:UpdatePlayer(playerID)
-  for _,w in r_ipairs(self.PlayerChangedList) do
-    w:PlayerChanged(playerID)
-  end
-  return
+	MessageProcessor:UpdatePlayer(playerID)
+	local _,_,spectator,teamID,_ = Spring.GetPlayerInfo(playerID)
```

Comment:
Use the localized version you've added, also no need to capture extra fake values past the ones you use:
```suggestion
	local _,_,spectator,teamID = spGetPlayerInfo(playerID)
```